tests/ftqc/test_pauli_tracker.py

```diff
@@ -186,3 +209,172 @@ def test_apply_clifford_ops_xz_value_err(self, clifford_op, xz):
             match="Please ensure xz are either 0 or 1.",
         ):
             _ = commute_clifford_op(clifford_op, xz)
+
+
+@flaky(max_runs=5)
+class TestOfflineCorrection:
+    """Tests for byproduct operation offline corrections."""
+
+    def _measurements_corrections(self, script, num_shots, raw_res):
+        """Helper function for correcting the measurement results with recorded MidMeasure results."""
+        meas_res = raw_res[0 : len(script.measurements)]
+
+        mid_meas_res = raw_res[len(script.measurements) :]
+
+        for i in range(num_shots):
+            mid_meas = [row[i] for row in mid_meas_res]
+            phase_cor = get_byproduct_corrections(script, mid_meas)
+            for j in range(len(script.measurements)):
+                meas_res[j][i] = meas_res[j][i] * phase_cor[j]
+
+        cor_res = []
+        for i in range(len(script.measurements)):
+            cor_res.append(np.sum(meas_res[i]) / num_shots)
+        return cor_res
+
+    def _get_mbqc_tape(self, ops, obs, num_shots):
+        """Helper function for creating a MBQC tape with a list of ops and obs."""
+        measurements = []
+        for ob in obs:
+            measurements.append(qml.sample(ob))
```

Comment:
at this point all the tapes should have `qml.sample(wires=all_measurement_wires)` as their output

Reply:
Done!